tests/build.gradle.kts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import com.google.protobuf.gradle.protobuf
+import com.google.protobuf.gradle.protoc
+import io.spine.internal.dependency.JUnit
+import io.spine.internal.dependency.Protobuf
+import io.spine.internal.dependency.Truth
+import io.spine.internal.gradle.Scripts
+import io.spine.internal.gradle.applyStandard
+
+@Suppress("RemoveRedundantQualifierName")
+plugins {
+    java
+    idea
+    with(io.spine.internal.dependency.Protobuf.GradlePlugin) {
+        id(id) version version
+    }
+}
+
+subprojects {
+    apply {
+        plugin("java")
+        plugin("idea")
+        plugin("com.google.protobuf")
+        from("$rootDir/../version.gradle.kts")
+        from(Scripts.testOutput(project))
+    }
+
+    repositories.applyStandard()
+
+    val generatedFiles = "$projectDir/generated"
+
+    protobuf {
+        generatedFilesBaseDir = generatedFiles
+        protoc {
+            artifact = Protobuf.compiler
+        }
+    }
+
+    tasks.getByName<Delete>("clean") {
+        delete.add(generatedFiles)
+    }
+
+    dependencies {
+        Protobuf.libs.forEach { implementation(it) }
+
+        JUnit.api.forEach { testImplementation(it) }
+        Truth.libs.forEach { testImplementation(it) }
+        testRuntimeOnly(JUnit.runner)
+    }
+
+    tasks.withType<Test> {
+        useJUnitPlatform()
+    }
+}
```

Comment:
I feel like this `tests` sub-project lacks a `README.md`. The flow is far from trivial, so a description would become handy for anyone testing their own extensions.